src/librustc_mir/borrow_check/nll/region_infer/mod.rs

```diff
@@ -1075,3 +1114,18 @@ impl<'gcx, 'tcx> ClosureRegionRequirementsExt<'gcx, 'tcx> for ClosureRegionRequi
         })
     }
 }
+
+trait CauseExt {
```

Comment:
~~Why an extension trait?~~ We have to wait until the next snapshot for `#![feature(arbitrary_self_types)]`

Reply:
Old habits die hard =)